File: frontend/src/microservices/operations/objects/UpdateHotelManagerProfilePlannerRequest.ts
// 本文件定义 operations 模块的 `UpdateHotelManagerProfilePlannerRequest`，作为planner 请求参数并提供 JSON 编解码。

export type UpdateHotelManagerProfilePlannerRequest = {
  managerId: string
  displayName: string
  email: string
  hotelName: string
  hotelLocation: string
}

export const updateHotelManagerProfilePlannerRequestFromJson = (json: string): UpdateHotelManagerProfilePlannerRequest =>
  JSON.parse(json) as UpdateHotelManagerProfilePlannerRequest

export const updateHotelManagerProfilePlannerRequestToJson = (value: UpdateHotelManagerProfilePlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/hotel/objects/UpdateHotelManagerProfilePlannerRequest.scala
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UpdateHotelManagerProfilePlannerRequest(
    managerId: String,
    displayName: String,
    email: String,
    hotelName: String,
    hotelLocation: String
)
object UpdateHotelManagerProfilePlannerRequest:
  given sourceEncoder: Encoder[UpdateHotelManagerProfilePlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[UpdateHotelManagerProfilePlannerRequest] = deriveDecoder
